File: Python/1550_three-consecutive-odds.py
class Solution:
    def threeConsecutiveOdds(self, arr: List[int]) -> bool:
        num_consecutive = 0

        for num in arr + [0]:
            if num % 2:
                num_consecutive += 1

                if num_consecutive == 3:
                    return True
            else:
                num_consecutive = 0

        return False

File: Elixir/1550_three-consecutive_odds.ex
defmodule Solution do
  @spec three_consecutive_odds(arr :: [integer]) :: boolean
  def three_consecutive_odds(arr) do
    Enum.reduce_while(arr, 0, fn num, acc ->
      cond do
        acc >= 3 -> {:halt, acc}
        rem(num, 2) == 1 -> {:cont, acc + 1}
        rem(num, 2) == 0 -> {:cont, 0}
      end
    end) >= 3
  end
end
